examples/find_all_posts_mentioning_github.py

```diff
@@ -1,25 +1,35 @@
-import twingly_search
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
 import datetime
 
+import twingly_search
+
+
 class SearchPostStream:
     def __init__(self, keyword, language=''):
         self.client = twingly_search.Client(user_agent="MyCompany/1.0")
-        self.query = self.client.query()
-        self.query.pattern = "sort-order:asc sort:published %s" % keyword
-        self.query.start_time = datetime.datetime.utcnow() - datetime.timedelta(hours=48)
-        self.query.language = language
+        self.q = self.client.query()
+        self.q.search_query = "%s sort-order:asc sort:published " % keyword
+        self.q.language = language
+        self.q.start_date = datetime.datetime.utcnow() - datetime.timedelta(weeks=10)
+        self.keyword = keyword
 
     def each(self):
         while True:
-            result = self.query.execute()
+            query_string = self.q.build_query_string()
+            result = self.client.execute_query(query_string)
```

Comment:
IMO Query and Client are too loosely coupled with each other.
I'd prefer to separate them so that Query doesn't depend on Client anymore.

It's quite strange when Query can execute itself that's why I've also marker Query.execute() as deprecated.